src/networking.c

```diff
@@ -281,6 +324,79 @@ get_address(const gint family, const void *const sin_addr)
 }
 
 /*!
+ * Count the number of valid bits in the subnet prefix
+ * Copied over from BSD
```

Comment:
@mcastelino @jodh-intel Adding BSD code to a GPL code base is fine, as long as we add the right labels to the file's header, i.e. something along those lines:

```
/*
 *  [GPL Header/Copyright]
 *  ...
 * This file incorporates work covered by the following copyright and  
 * permission notice:
 * [BSD license and copyright for the included code]
 */
```